### PR TITLE
Add Autocomplete edit field option

### PR DESCRIPTION
**Pull request by matt-eric**
## Related Issue

#1410 

## Description

Enables `Autocomplete` inputs to be used as edit fields.

image

## Impacted Areas in Application

m-table-edit-field.js

## Additional Notes

An example of how the `Autocomplete` is added to the `columns` prop in the `<MaterialTable/>` component is here:

image

The `autocomplete` key is given an array to be used as the `options` for the `Autocomplete` input element. The `freeSolo` prop can optionally be configured here as well.

The `fontSize` for the other editable input components has been adjusted to the default `fontSize` (from 13 to 16) because there were issues with adjusting the `fontSize` of the `Autocomplete`. The `fontSize` for table cells in the non-editable state remains the same.

Please let me know if there are any comments or questions. I'm able to make changes if requested.

### CODE REVIEW

**Comment by matt-eric** [comment]
Converting to draft because I realize that the Autocomplete fields are only filtering results on keyboard input when a **new** row has been added. After the Check icon is clicked, when a user clicks to edit the row and inputs a new value into the Autocomplete TextField, the filtering no longer happens when typing. I will fix and convert back.

File: src/components/m-table-edit-field.js [hunk 1]
```diff
@@ -3,25 +3,62 @@ import TextField from '@material-ui/core/TextField';
...
+    });
+  }
+
+  renderAutocompleteField() {
+    return (
+      <Autocomplete
+        {...this.getProps()}
+        clearOnEscape
+        options={this.props.columnDef.autocomplete}
+        freeSolo={this.props.columnDef.freeSolo}
```
Comment (mbrn) [inline src/components/m-table-edit-field.js:38]: can you rename freeSolo as autocompleteFreeSolo, please

File: src/components/m-table-edit-field.js [hunk 2]
```diff
@@ -3,25 +3,62 @@ import TextField from '@material-ui/core/TextField';
...
+        value={this.state.secondaryValue || ''}
+        onChange={(event) => this.handleChangeForSecondaryInput(event)}
+        renderInput={(params) =>
+          <TextField
+            {...params}
+            placeholder={this.props.columnDef.title}
```
Comment (mbrn) [inline src/components/m-table-edit-field.js:44]: After merging a related pr, you can replace this with

```js
this.props.columnDef.editPlaceholder || this.props.columnDef.title

File: src/components/m-table-edit-field.js [hunk 3]
```diff
@@ -87,9 +120,6 @@ class MTableEditField extends React.Component {
           onChange={this.props.onChange}
           clearable
           InputProps={{
-            style: {
```
Comment (mbrn) [inline src/components/m-table-edit-field.js:90]: Why did you removed this?

**Comment by mbrn** [comment]
Hi @matt-eric 

Thanks for your effort. I requested a few changes. And please resolve conflicts before pushing. Thanks again

**Comment by mbrn** [comment]
I will close the PR. You can reopen it after completing change requests.

**Comment by matt-eric** [comment]
Moved to https://github.com/mbrn/material-table/pull/2134/files

**Pull request closed without merging**